pysmt/formula.py

```diff
@@ -939,9 +941,14 @@ def normalize(self, formula):
               obtain f_b that is the formula f_a expressed on the
               FormulaManager b : f_b = b.normalize(f_a)
         """
-        # TODO: name clash with formula normalization
-        # TODO: Move this out of the manager and into ad-hoc function
-        normalizer = IdentityDagWalker(self.env)
+        # TODO: Move this to environment.py
+        class FormulaContextualizer(IdentityDagWalker):
+            def walk_symbol(self, formula, **kwargs):
```

Comment:
Actually, also ArrayValues have the index type as payloads and need to be normalized